nmstatectl/nmstatectl.py

```diff
@@ -50,12 +54,20 @@ def setup_subcommand_set(subparsers):
 
 
 def show(args):
-    print(json.dumps(netinfo.show(), indent=4, sort_keys=True,
-                     separators=(',', ': ')))
+    state = netinfo.show()
+    if args.yaml:
+        print(yaml.dump(state, default_flow_style=False))
+    else:
+        print(json.dumps(state, indent=4, sort_keys=True,
+                         separators=(',', ': ')))
 
 
 def apply(args):
     with open(args.file) as f:
-        state = json.load(f)
+        try:
+            state = json.load(f)
+        except ValueError:
+            f.seek(0)
```

Comment:
We want to add stdin support, right? Can we seek 0 from stdin? If not, it might be better to load input and than use loads.

Reply:
thx, not sure it is a priority but seeking from stdin, so here is an update